### PR TITLE
Allow ordering of results

### LINKED ISSUE #74: Allow ordering of results
Implement SQLs `ORDER BY (DESC)` as allowed query parameters

### PR DESCRIPTION
**Pull request by markharley**
Closes #74

Allows queries to specify SQL `ORDER BY (DESC)`.

Fix a bug in using `limit` and `offset`. Failed when parsing the
limit/offset from JSON as we'd end up with a string. Added an
IntegerValidator to coerce this to int.

### CODE REVIEW

**Comment by markharley** [comment]
I'll have to update this sorry. Found a bug

File: tests/example/models.py [hunk 1]
```diff
@@ -34,11 +35,17 @@ class Task(TaskAdd):
...
 @dataclass
 class TaskQuery:
+    title: str = data_field()
     done: bool = data_field()
     type: TaskType = enum_field(TaskType)
     severity: int = decimal_field(ops=('lt', 'le', 'gt', 'ge', 'ne'))
 
 
+@dataclass
+class TaskOrderableQuery(TaskQuery, orderable('title'), Query):
```
Comment (ConorStevenson) [inline tests/example/models.py]: Weird, but I guess this works

File: openapi/data/fields.py [hunk 2]
```diff
@@ -234,16 +234,11 @@ def __call__(self, field, value, data=None):
...
 
-    def __init__(self, min_value=None, max_value=None, precision=None):
+    def __init__(self, min_value=None, max_value=None):
         self.min_value = min_value
         self.max_value = max_value
-        self.precision = precision
 
     def __call__(self, field, value, data=None):
-        try:
-            value = round(value, self.precision)
```
Comment (lsbardel) [inline openapi/data/fields.py]: we still need to test if it is a valid number right?
```python
value = float(value)
```

Comment (markharley) [reply openapi/data/fields.py]: I've moved this out of `NumberValidator` and into `DecimalValidator` now so we're doing a `Decimal` conversion before round

**Pull request merged**